multipass-addressing/src/main/scala/com/precisely/bigdata/sample/spark/MultipassAddressing.scala

```diff
@@ -0,0 +1,181 @@
+  }
+}
+
+/*
+ * A custom Executor that contains the logic to perform verify and then multipass geocode
+ */
+class CustomExecutor extends AddressingExecutor {
+
+  import com.precisely.addressing.v1.model.Response
+  import com.precisely.addressing.v1.{Addressing, Preferences}
+
+  // set preference to return all fields and Match Mode to Relaxed
+  private val relaxedGeocoderPreferences: Preferences = new PreferencesBuilder()
+    .withReturnAllInfo(true)
+    .withMatchMode("RELAXED")
+    .build()
+
+  override def execute(input: RequestInput, preferences: Option[Preferences], addressing: Addressing): Response = {
+
+    val verifyResponse = addressing.verify(input.requestAddress(), preferences.orNull)
+
+    var multiLineRequest: RequestAddress = input.requestAddress()
+
+    var singleLine = input.requestAddress().getStreet + " " +
+      input.requestAddress().getCity + " " +
+      input.requestAddress().getAdmin1 + " " +
+      input.requestAddress().getPostalCode
+
+    if (CollectionUtils.isNotEmpty(verifyResponse.getResults)) {
+      val verifyAddress = verifyResponse.getResults.get(0).getAddress
+      //  create multiLine request from verify Response
+      multiLineRequest = new RequestAddress()
+      multiLineRequest.setStreet(verifyAddress.getStreet)
+      multiLineRequest.setAddressNumber(verifyAddress.getAddressNumber)
+      multiLineRequest.setAdmin1(verifyAddress.getAdmin1.getLongName)
+      multiLineRequest.setAdmin2(verifyAddress.getAdmin2.getLongName)
+      multiLineRequest.setCountry(verifyAddress.getCountry.getIsoAlpha3Code)
+      multiLineRequest.setCity(verifyAddress.getCity.getLongName)
+      multiLineRequest.setPostalCode(verifyAddress.getPostalCode)
+
+      // create singleline request from verify Response
+      singleLine = verifyAddress.getFormattedAddress + " " + verifyAddress.getCountry.getIsoAlpha3Code
+    }
+
+    val multiLineResponse = addressing.geocode(multiLineRequest, relaxedGeocoderPreferences)
+
+    if (CollectionUtils.isNotEmpty(multiLineResponse.getResults) && isAddressLevelMatch(multiLineResponse)) {
+      return multiLineResponse
+    }
+
+    val singleLineRequest: RequestAddress = new RequestAddress()
+    singleLineRequest.setAddressLines(Collections.singletonList(singleLine))
+
+    val singleLineResponse = addressing.geocode(singleLineRequest, relaxedGeocoderPreferences)
+
+    if (CollectionUtils.isNotEmpty(singleLineResponse.getResults) && isAddressLevelMatch(multiLineResponse)) {
+      return singleLineResponse
+    }
+
+    if (CollectionUtils.isNotEmpty(singleLineResponse.getResults) && CollectionUtils.isNotEmpty(multiLineResponse.getResults) &&
+      multiLineResponse.getResults.get(0).getScore > singleLineResponse.getResults.get(0).getScore) {
+      return multiLineResponse
+    }
+
+    if (singleLineResponse != null) singleLineResponse else multiLineResponse
+
+  }
+
+  def isAddressLevelMatch(response: Response): Boolean = {
```

Comment:
lets clean up the signature here and take "Result" (a single address) rather than Response

Reply:
ok